build.sbt

```diff
@@ -107,9 +107,8 @@ testScalastyle := scalastyle.in(Test).toTask("").value
 def getVersion(version: String): String = {
     version.split("\\.").toList match {
         case major :: minor :: rest =>
-          if (rest.contains("0-SNAPSHOT")) s"$major.${minor.toInt - 1}.0"
-          else if (rest.contains("SNAPSHOT")) s"$major.$minor.${rest.head.toInt - 1}"
-          else s"$major.$minor.0"
+          if (rest.head.startsWith("0")) s"$major.${minor.toInt - 1}.0"
+          else s"$major.$minor.${rest.head.toInt - 1}"
```

Comment:
read.head? this wont accommodate double-digit patch versions like 0.7.10... right? 
on second thought, in this case, lets just compare with `"$major.$minor.0"` 

So the logic is if it is currently or going to be minor release (`x.y.0...`) then use the previous minor release `x.y-1.0`. Otherwise it is patch release, and then use the original minor release corresponding to it (`x.y.0`).

Reply:
Hello @tdas ,
It will accomodate with double-digit patch versions, ```rest``` is a list which contains the remaining elements of the split operation (in that case, for 0.7.10, rest will be equal to List(10))
Here's a test I did in the scala REPL : 
<img width="374" alt="Capture d’écran 2020-06-22 à 23 12 20" src="https://user-images.githubusercontent.com/4971795/85335785-dc004a00-b4dd-11ea-8961-6fa4f219fd00.png">
(Maybe we should move this method somewhere else and write an appropriate test for it)